api/devices/schema.py

```diff
@@ -15,12 +19,17 @@ class Meta:
 class CreateDevice(graphene.Mutation):
     class Arguments:
         name = graphene.String(required=True)
+        room_id = graphene.Int(required=True)
         device_type = graphene.String(required=True)
         location = graphene.String()
-        resource_id = graphene.Int(required=True)
     device = graphene.Field(Devices)
 
+    @Auth.user_roles('Admin')
     def mutate(self, info, **kwargs):
+        room_location = location_join_room().filter(RoomModel.id == kwargs['room_id']).first()  # noqa: E501
```

Comment:
Avoid the use of noqa in the codebase, ensure that you limit the number of characters to 79 per line to get rid of flake8 errors.

Reply:
Thanks, worked on this feedback